app/controllers/service_controller.rb

```diff
@@ -33,9 +33,23 @@ def button
       service_retire
     when 'service_retire_now'
       service_retire_now
+    when "custom_button"
+      @display == 'generic_objects' ? generic_object_custom_buttons : custom_buttons
+    else
+     add_flash("Invalid button action", :error)
```

Comment:
This looks better.  Does "Unsupported button action" sound better to you?  I don't know.    Do we need to see the `params[:pressed]` value here?  

Can you add translation support here? 

```ruby
add_flash(_("Invalid button action!"), :error)
```